### DISCUSSION TITLE
How to recover from failure in switching lazy loading routes?

### DISCUSSION THREAD

**Discussion by mengdu** [vuejs/router#1864] [Help and Questions] [answered]
### What problem is this solving

When use lazy loading routing the new version is redeployed in the background, switching to lazy loading routing on the current browsing page will fail (the routing page is not found).

How to capture this error and display it to inform the user that they need to refresh the page to load the latest version?

I know it can be solved this way:

```js
import ErrorBoundary from './ErrorBoundary.vue'
const Page = () => import('./pages/xxx.vue').catch(err => ErrorBoundary)
```

But it's not the optimal solution.

### Proposed solution

The routing library supports this error capture definition error recover.

**Comment by posva** [comment, marked as answer]
use the `router.onError()` hook to detect these errors and handle it the way you prefer, e.g. reloading the page
